google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java

```diff
@@ -0,0 +1,200 @@
+              && response
+                  .bodyString()
+                  .map(RabbitMQCredentialTester::bodyContainsRabbitMQElements)
+                  .orElse(false);
+      url = buildTargetUrl(networkService, "api/overview");
+      response = httpClient.send(get(url).withEmptyHeaders().build());
+      canAcceptByCustomFingerprint =
+          canAcceptByCustomFingerprint
+              && response.headers().get("www-authenticate").isPresent()
+              && response.headers().get("www-authenticate").get().equals(RABBITMQ_WWW_HEADER);
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", url);
+      return false;
+    }
+    return canAcceptByCustomFingerprint;
+  }
+
+  @Override
+  public ImmutableList<TestCredential> testValidCredentials(
+      NetworkService networkService, List<TestCredential> credentials) {
+
+    return credentials.stream()
+        .filter(cred -> isCredentialValid(networkService, cred))
+        .collect(toImmutableList());
+  }
+
+  private boolean isCredentialValid(NetworkService networkService, TestCredential credential) {
+    var url = buildTargetUrl(networkService, "api/whoami");
+    try {
+      logger.atInfo().log(
+          "url: %s, username: %s, password: %s",
+          url, credential.username(), credential.password().orElse(""));
+      HttpResponse response = sendRequestWithCredentials(url, credential);
+
+      return response.status().isSuccess()
+          && response
+              .bodyString()
+              .map(RabbitMQCredentialTester::bodyContainsSuccessfulLoginElements)
+              .orElse(false);
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", url);
+      return false;
+    }
+  }
+
+  private HttpResponse sendRequestWithCredentials(String url, TestCredential credential)
+      throws IOException {
+
+    var headers =
+        HttpHeaders.builder()
+            .addHeader(
+                "Authorization",
+                "Basic "
+                    + Base64.getEncoder()
+                        .encodeToString(
+                            (credential.username() + ":" + credential.password().orElse(""))
+                                .getBytes(UTF_8)))
+            .build();
+
+    return httpClient.send(get(url).setHeaders(headers).build());
+  }
+
+  /**
+   * A successful authenticated request to the /api/whoami endpoint returns a JSON with at least the
+   * following keys: {"name":"username","tags":["roles"]}
+   */
+  private static boolean bodyContainsSuccessfulLoginElements(String responseBody) {
+    try {
+      JsonObject response = JsonParser.parseString(responseBody).getAsJsonObject();
+
+      if (response.has("name") && response.has("tags")) {
+        logger.atInfo().log("Successfully logged in to RabbitMQ Management Portal");
+        return true;
+      } else {
+        return false;
+      }
+    } catch (Exception e) {
```

Comment:
This catch block catches `Exception`, but can only catch unchecked exceptions. Consider catching RuntimeException (or something more specific) instead so it is more apparent that no checked exceptions are being handled.